src/ptvsd/wrapper.py

```diff
@@ -1582,7 +1582,16 @@ def on_stackTrace(self, request, args):
             pydevd_request,
             is_json=True)
 
-        stackFrames = resp_args['body']['stackFrames']
+        levels = int(args.get('levels', 0))
+        if levels == 0:
+            stackFrames = resp_args['body']['stackFrames']
+        else:
+            try:
+                start = int(args.get('startFrame', 0))
+                end = start + levels
+                stackFrames = resp_args['body']['stackFrames'][start:end]
+            except:
+                stackFrames = []
```

Comment:
The DAP spec says that levels is the "maximum number of frames to return", so it's an upper bound, not a request for a specific number - but this will return an empty list if `end` is out of bounds. OTOH it will return an empty list instead of reporting an error on invalid `start`, because of the catch-all except.

 I think this needs to be more permissive on `end`, and just allow exception to propagate on invalid `start`; something like:
```py
        stackFrames = resp_args['body']['stackFrames']
        if levels > 0:
                start = int(args.get('startFrame', 0))
                end = min(start + levels, len(stackFrames))
                stackFrames = stackFrames[start:end]
```